### PR TITLE
Fix build errors of issue #46

### LINKED ISSUE #46: Build status of master is failing
Locally all tests pass:
imagen
But on travis they fail...

### PR DESCRIPTION
**Pull request by alejgh**
Work done:
* I have restored the sbt version to 1.3.7. With that the following error is fixed: `The compiler bridge sources org.scala-sbt:compiler-bridge_2.13:1.1.1:compile could not be retrieved`.
* I have modified the project to be compatible with both Scala 2.12.10 and 2.13.1. This allows us to make two different Travis builds, one when the coverage is run (2.12.10) and the other one where just tests are run (2.13.1)
* I have submitted a temporal fix for the error where one test would fail when executing tests from the command line. The problem comes from the MemorySymbolTableUnitTest class. When running the test suite the tests from this class are executed before the IdentificationWalker tests, and the state of the MemoryErrorHandler was not restored, containing an error from the MemorySymbolTable tests. **In theory**, the state should be restored with the *before* method, but that was not the case. For now I substituted the before method for an after one, so after each test in the MemorySymbolTable suite the state is restored. This works right now, but we should open a new issue to look a the cause of this problem and propose a better solution.

Closes #46

### CODE REVIEW

File: test/unit/compiler/semantic/IdentificationWalkerTest.scala [hunk 1]
```diff
@@ -33,11 +33,8 @@ class IdentificationWalkerTest extends AnyFunSuite with BeforeAndAfter {
   final val logger = Logger[IdentificationWalkerTest]
 
   // In order to be sure that on each test case we do not have data from previous tests.
-  before {
-    logger.debug("Restoring memory symbol table.")
```
Comment (thewilly) [inline test/unit/compiler/semantic/IdentificationWalkerTest.scala:37]: What happened to the logger?

File: test/unit/compiler/MemorySymbolTableUnitTest.scala [hunk 2]
```diff
@@ -48,7 +48,7 @@ class MemorySymbolTableUnitTest extends AnyFunSuite with BeforeAndAfter {
   val shape = new ShapeDeclaration("example", 4,1,
     new PrefixInvocation("example", 4,2, "this is my shape name", prefix), null)
 
-  before {
+  after {
```
Comment (thewilly) [inline test/unit/compiler/MemorySymbolTableUnitTest.scala:51]: Was any problem with the original before impl?

Comment (alejgh) [reply test/unit/compiler/MemorySymbolTableUnitTest.scala:51]: yes, I will document everything later on in the issue, this is still WIP

Comment (thewilly) [reply test/unit/compiler/MemorySymbolTableUnitTest.scala:51]: okay thanks!

Comment (alejgh) [reply test/unit/compiler/MemorySymbolTableUnitTest.scala:51]: I have updated the comment

**Comment by alejgh** [comment]
Right now the only thing that bothers me is that Travis is making a build just for Scala 2.13.1 and not for both 2.13.1 and 2.12.10. Do you know what could be the problem?

**Comment by alejgh** [comment]
I would like to check the build with 2.12.10 (where the coverageReport is done) before merging this.

**Comment by thewilly** [comment]
> 
> 
> I would like to check the build with 2.12.10 (where the coverageReport is done) before merging this.

Yes I've been looking the travis generated logs and it is skiping the 2.12.10 version.. Would like to fix it before merging too.

**Comment by alejgh** [comment]
I think I got it, the problem is that we are using the "matrix" functionality to run a Python build too. I separated the 2.12.10 and 2.13.1 builds, lets see if it works.

**Comment by thewilly** [comment]
If it works you can have the rest of the week free >.< (It's friday)

**Comment by alejgh** [comment]
> If it works you can have the rest of the week free >.< (It's friday)

Nah, the 2.12.10 build is failing. This is going to be a tough one...

**Comment by alejgh** [comment]
Seems to be working now.

File: test/unit/compiler/semantic/IdentificationWalkerTest.scala (hunk 1 above)
Comment (alejgh) [reply test/unit/compiler/semantic/IdentificationWalkerTest.scala:37]: I restored the logging messages in the following commit, my bad.

**Pull request merged**